packages/flutter_tools/test/general.shard/mdns_discovery_test.dart

```diff
@@ -991,13 +1013,15 @@ class FakeMDnsClient extends Fake implements MDnsClient {
     this.txtResponse = const <String, List<TxtResourceRecord>>{},
     this.ipResponse = const <String, List<IPAddressResourceRecord>>{},
     this.osErrorOnStart = false,
+    this.socketErrorOnStart = false
   });
 
   final List<PtrResourceRecord> ptrRecords;
   final Map<String, List<SrvResourceRecord>> srvResponse;
   final Map<String, List<TxtResourceRecord>> txtResponse;
   final Map<String, List<IPAddressResourceRecord>> ipResponse;
   final bool osErrorOnStart;
+  final bool socketErrorOnStart;
```

Comment:
Ultra pendantic nit: errors in Dart are unrecoverable exceptions that shouldn't be caught (see [this doc](https://dart.dev/language/error-handling)). I'd consider renaming this to `socketExceptionOnStart`